### ISSUE TITLE
Uncaught Reference Error in node

### ISSUE THREAD

**Issue by KingCode** [AlexSp3/Basenumber.js#2]
I fired a node process and `require`ed basenumber.js. It seems line 490 declaring `const P = this` should cause P to be visible from within the block checking for module.exports at line 1091.
This library is a great and welcome addition to do the usual math stuff with BigInt - looking forward to see it working!
Thanks..
(I also tried the  minified version with same results)
bash3.2$ node
Welcome to Node.js v17.0.1.
Type ".help" for more information.
> const Base = require('./lib/basenumber.js');
const Base = require('./lib/basenumber.js');
Uncaught ReferenceError: P is not defined
    at /<path..>/lib/basenumber.js:1093:13
    at Object.<anonymous> (/<path>/lib/basenumber.js:1109:3)
    at Module._compile (node:internal/modules/cjs/loader:1095:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1147:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
>

**Comment by AlexSp3** [comment]
Thanks for yours encouraging comments!
Yes I see the error and it seems to be an scope issue with the `P` var. I will make a quick fix in the next days. I'm also planning to add full bigint compatibility soon, accepting a bigint in the constructor.

**Comment by KingCode** [comment]
Hello Alexandro,

Full BigInt compatibility would be awesome! All other libraries I have
tried seem to be lossy w.r.t. big ints. - if a big int has 1M digits, all
of them should be preserved :)

JF

**Comment by AlexSp3** [comment]
Hello King,
I have just fixed the scope error you have mentioned so it should work ok now. I also added the posibility of pass a base-10 BigInt to the constructor!
As you said, native BigInt doesn't seem to have a length limit rather than your pc memory, so they shouldn't be lossy digits. However, there are some operations that have a precision limit such as decimals  `pow` ,because it is based on `LN2` constant, which has 1025 digits.
